pkg/shipper/shipper.go

```diff
@@ -383,7 +389,23 @@ func (s *Shipper) upload(ctx context.Context, meta *metadata.Meta) error {
 	if err := meta.WriteToDir(s.logger, updir); err != nil {
 		return errors.Wrap(err, "write meta file")
 	}
-	return block.Upload(ctx, s.logger, s.bucket, updir, s.hashFunc)
+
+	err := block.Upload(ctx, s.logger, s.bucket, updir, s.hashFunc)
+	if err != nil {
+		return errors.Wrap(err, "while upploading the block")
+	}
```

Comment:
I would suggest inlining it (also typo in error message): 

```suggestion
	if err := block.Upload(ctx, s.logger, s.bucket, updir, s.hashFunc); if err != nil {
		return errors.Wrap(err, "while uploading the block")
	}
```

Rationales: https://thanos.io/tip/contributing/coding-style-guide.md/#avoid-variable-shadowing 🤗

Reply:
Agree, will change and keep this in mind for the future!